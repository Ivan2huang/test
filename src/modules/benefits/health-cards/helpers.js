/* eslint-disable import/prefer-default-export */
export const getQRCodeXML = ({
  fullName,
  membershipNumber,
  certificateNumber,
  insurerCode,
  insurerName,
  insurerType,
  policyNumber,
  expiryDate,
}) =>
  [
    '<?xml version="1.0"?>',
    '<medical_card>',
    '<Card_type>',
    insurerType,
    '</Card_type>',
    '<insurance_company_code>',
    insurerCode,
    '</insurance_company_code>',
    '<insurance_company_name>',
    insurerName,
    '</insurance_company_name>',
    '<insured_name>',
    fullName,
    '</insured_name>',
    '<policy_no>',
    policyNumber,
    '</policy_no>',
    '<certificate_no>',
    certificateNumber,
    '</certificate_no>',
    '<membership_no>',
    membershipNumber,
    '</membership_no>',
    '<expiry_date>',
    expiryDate,
    '</expiry_date>',
    '</medical_card>',
  ].join('');
