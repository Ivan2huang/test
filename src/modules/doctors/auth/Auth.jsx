import React, { useEffect } from 'react';
import { navigateTo } from '../../../helpers/helpers';
import { FIND_DOCTOR_PATHNAME } from '../constant';

const Auth = () => {
  useEffect(() => {
    navigateTo(FIND_DOCTOR_PATHNAME);
  }, []);
  return <></>;
};
export default Auth;
