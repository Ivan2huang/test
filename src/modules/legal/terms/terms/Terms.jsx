/* eslint-disable react/no-danger */
import React, { useEffect, useState } from 'react';
import * as PropTypes from 'prop-types';
import Head from 'next/head';
import { Box, Button, Tabs, Tab } from '@material-ui/core';
import { withTheme, makeStyles } from '@material-ui/styles';
import { createIntl, createIntlCache, RawIntlProvider } from 'react-intl';
import Checkbox from '@material-ui/core/Checkbox';
import FormControlLabel from '@material-ui/core/FormControlLabel';

import Router, { useRouter } from 'next/router';

import Typography from '../../../../uiComponents/Typography';
import ButtonGroup from '../../../../uiComponents/ButtonGroup';
import {
  formatMessage,
  isValidLanguageCode,
} from '../../../../helpers/helpers';
import paths from '../../../../helpers/paths';
import { getCookie } from '../../../../helpers/auth';
import { LOGGED_IN } from '../../../../constants/auth';
import breakpoints from '../../../../themes/breakpoints';
import CONFIG from '../../../../constants/config';
import getLocale from '../../../../i18n/getLocale';
import { messages } from '../../../../i18n/lang';
import { LOCALE_STORAGE } from '../../constants';
import { DEFAULT_LANGUAGE_NAME_MAPPING } from '../../../../constants/types';

const navigateToPreviousPage = Router.back;
const cache = createIntlCache();

const getDescriptionStyles = props => {
  const { theme, doesShowTab } = props;
  const minHeight = doesShowTab ? theme.spacingX(90) : theme.spacingX(100);
  const height = doesShowTab
    ? `calc(81vh - ${theme.spacingX(51)})`
    : `calc(100vh - ${theme.spacingX(51)})`;
  const minHeightMdUp = doesShowTab ? theme.spacingX(125) : theme.spacingX(125);
  const heightMdUp = `calc(100vh - ${theme.spacingX(120)})`;

  return {
    minHeight,
    height,
    overflowY: 'scroll',
    textAlign: 'left',
    paddingRight: theme.spacingX(3),
    [breakpoints.up('md')]: {
      minHeight: minHeightMdUp,
      height: heightMdUp,
    },
  };
};

const useStyles = makeStyles({
  description: getDescriptionStyles,
  firstTab: props => ({
    marginRight: props.theme.spacingX(10),
  }),
  tabContainer: props => ({
    paddingTop: props.theme.spacingX(2),
    [breakpoints.up('md')]: {
      paddingTop: props.theme.spacingX(10),
    },
  }),
  link: props => ({
    color: props.theme.hyperlink,
  }),
});

const renderAcceptCheckbox = (classes, isTermsAccepted, onChange, intl) => {
  return (
    <Box>
      <FormControlLabel
        className={classes.root}
        control={
          // eslint-disable-next-line react/jsx-wrap-multilines
          <Checkbox
            value={isTermsAccepted}
            color="primary"
            data-testid="checkbox-terms"
            onChange={e => onChange(e.target.checked)}
          />
        }
        label={
          // eslint-disable-next-line react/jsx-wrap-multilines
          <Typography type="style-6">
            {formatMessage(
              intl,
              'legal.terms.agree.label',
              'I Accept the terms and conditions',
            )}
          </Typography>
        }
      />
    </Box>
  );
};

const renderPromotionalLink = (intl, classes) => {
  return (
    <a href="/legal/news-letter" className={classes.link}>
      {formatMessage(
        intl,
        'legal.edm.promotional.link',
        'promotional materials',
      )}
    </a>
  );
};

const renderEdmCheckbox = (classes, isEdmAccepted, onChange, intl) => {
  return (
    <Box mb={{ sm: 5, md: 10 }}>
      <FormControlLabel
        className={classes.root}
        control={
          // eslint-disable-next-line react/jsx-wrap-multilines
          <Checkbox
            value={isEdmAccepted}
            color="primary"
            data-testid="checkbox-edm"
            onChange={e => onChange(e.target.checked)}
          />
        }
        label={
          // eslint-disable-next-line react/jsx-wrap-multilines
          <Typography type="style-6">
            {formatMessage(
              intl,
              'legal.edm.agree.label',
              '(Optional) I do not want to receive health, wellness or insurance {promotional} (by selecting this option, you will not receive our exclusive offers)',
              {
                promotional: renderPromotionalLink(intl, classes),
              },
            )}
          </Typography>
        }
      />
    </Box>
  );
};

const Terms = ({
  acceptTerms,
  theme,
  termLocales,
  termsConditions,
  getTermsConditions,
  alreadyAcceptedTerms,
}) => {
  const { query } = useRouter();
  const userProvidedLocale = isValidLanguageCode(query.lang)
    ? query.lang
    : undefined;
  const locale =
    userProvidedLocale || localStorage.getItem(LOCALE_STORAGE) || getLocale();
  const [selectedTab, setSelectedTab] = useState(locale);
  const [isTermsAccepted, setIsTermsAccepted] = useState(false);
  const [isEdmAccepted, setIsEdmAccepted] = useState(false);

  const doesShowTab = Object.keys(termsConditions).length > 1;
  const isLoggedIn = getCookie(LOGGED_IN);

  const intl = createIntl(
    {
      locale: selectedTab,
      messages: messages[selectedTab],
    },
    cache,
  );

  useEffect(() => {
    getTermsConditions(false, termLocales);
  }, []);

  const selectTab = value => {
    setSelectedTab(value);
    localStorage.setItem(LOCALE_STORAGE, value);
  };

  const onChange = value => {
    setIsTermsAccepted(value);
  };

  const onEdmChange = value => {
    setIsEdmAccepted(value);
  };

  const logout = () => {
    window.localStorage.clear();
    window.open(paths.common.logout, '_self');
  };

  const classes = useStyles({ theme, doesShowTab });

  const displayGoBackButton = alreadyAcceptedTerms || !isLoggedIn;

  const onAccept = () => {
    acceptTerms(isEdmAccepted);
  };

  return (
    <RawIntlProvider>
      <Box>
        <Head>
          <title>
            {formatMessage(intl, 'legal.terms.header', 'Terms & conditions')}
          </title>
        </Head>
        <Typography type="style-2">
          {formatMessage(intl, 'legal.terms.header', 'Terms & conditions')}
        </Typography>
        {doesShowTab && (
          <Tabs
            classes={{ root: classes.tabContainer }}
            value={selectedTab}
            onChange={(_, value) => selectTab(value)}
          >
            {CONFIG.supportedLanguages.map((spLang, spIndex) => {
              const Label = () => {
                let label = DEFAULT_LANGUAGE_NAME_MAPPING[spLang];
                if (spLang === CONFIG.locales.ENGLISH) {
                  label = label.replace('UK', '').trim();
                }
                return <Typography type="style-4">{label}</Typography>;
              };

              return (
                <Tab
                  data-testid={`${spLang}-tab`}
                  value={spLang}
                  disableRipple
                  disableFocusRipple
                  classes={{ root: spIndex === 0 ? classes.firstTab : '' }}
                  label={<Label />}
                />
              );
            })}
          </Tabs>
        )}
        <Box
          mt={6}
          mb={{ xs: 5, md: 10 }}
          classes={{ root: classes.description }}
        >
          <Typography type="style-6">
            <span
              dangerouslySetInnerHTML={{
                __html: termsConditions[selectedTab],
              }}
            />
          </Typography>
        </Box>
        {!displayGoBackButton && (
          <>
            {renderAcceptCheckbox(classes, isTermsAccepted, onChange, intl)}
            {renderEdmCheckbox(classes, isEdmAccepted, onEdmChange, intl)}
          </>
        )}
        {displayGoBackButton ? (
          <Button
            data-testid="btn-back-to-previous-page"
            variant="contained"
            color="primary"
            onClick={navigateToPreviousPage}
          >
            {formatMessage(intl, 'error.button.goback', 'Go back')}
          </Button>
        ) : (
          <ButtonGroup>
            <Button
              data-testid="btn-accept-terms"
              variant="contained"
              color="primary"
              onClick={onAccept}
              disabled={!isTermsAccepted}
            >
              {formatMessage(intl, 'legal.terms.agree', 'Agree and continue')}
            </Button>
            <Button
              data-testid="btn-logout"
              variant="outlined"
              color="secondary"
              onClick={logout}
            >
              {formatMessage(
                intl,
                'legal.terms.disagree',
                'Disagree and logout',
              )}
            </Button>
          </ButtonGroup>
        )}
      </Box>
    </RawIntlProvider>
  );
};

Terms.propTypes = {
  theme: PropTypes.shape({}).isRequired,
  termLocales: PropTypes.arrayOf(PropTypes.string),
  termsConditions: PropTypes.shape({}).isRequired,
  getTermsConditions: PropTypes.func.isRequired,
  alreadyAcceptedTerms: PropTypes.bool.isRequired,
  acceptTerms: PropTypes.func.isRequired,
};

Terms.defaultProps = {
  termLocales: [CONFIG.locales.ENGLISH],
};

export default withTheme(Terms);
