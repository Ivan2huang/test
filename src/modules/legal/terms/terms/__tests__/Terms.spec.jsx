/* eslint-disable func-names */
/* eslint-disable react/prop-types,no-undef */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Router from 'next/router';
import { IntlProvider } from 'react-intl';

import { IntlContext } from '../../../../../i18n/withIntlProvider';
import Terms from '../Terms';
import withTheme from '../../../../../themes/withThemeProvider';
import { getCookie } from '../../../../../helpers/auth';

jest.mock(
  '../../../../../uiComponents/Typography',
  () => ({ children, ...rest }) => (
    <div>
      Typography component
      <span>{mockPropsCapture(rest)}</span>
      <span>{children}</span>
    </div>
  ),
);

jest.mock('next/router', () => ({
  back: jest.fn(),
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    };
  },
}));

jest.mock('../../../../../helpers/helpers', () => ({
  formatMessage: jest.fn((intl, id, defaultMessage) => defaultMessage),
  isValidLanguageCode: jest.fn(() => false),
}));

jest.mock('../../../../../helpers/paths', () => ({
  common: {
    logout: '/logout',
  },
}));

jest.mock('../../../../../helpers/auth', () => ({
  getCookie: jest.fn(),
}));

jest.mock('../../../../../i18n/getLocale', () => ({
  __esModule: true,
  default: jest.fn(() => 'zh-HK'),
}));

jest.mock('../../../../../constants/config', () => ({
  supportedLanguages: ['en-HK', 'zh-HK'],
  locales: {
    ENGLISH: 'en-HK',
  },
}));

afterEach(() => {
  jest.clearAllMocks();
});

const setLocale = jest.fn();

describe('Terms component', () => {
  const termsConditions = { 'zh-HK': 'Terms and conditions' };
  const defaultProps = {
    termsConditions,
    getTermsConditions: jest.fn(),
    acceptTerms: jest.fn(),
    alreadyAcceptedTerms: false,
    preferredLocale: 'en-HK',
  };

  const setUp = (props = defaultProps) => {
    const Component = withTheme(Terms);
    return render(
      <IntlContext.Provider value={setLocale}>
        <IntlProvider locale="en-HK" messages={{}}>
          <Component {...props} />
        </IntlProvider>
      </IntlContext.Provider>,
    );
  };

  it('should match snapshot when user is logged in and has not accepted terms', () => {
    getCookie.mockReturnValue(true);
    const { container } = setUp();

    expect(container).toMatchSnapshot();
    expect(defaultProps.getTermsConditions).toHaveBeenCalled();
  });

  it('should call getTermsConditions on mount', () => {
    setUp();

    expect(defaultProps.getTermsConditions).toHaveBeenCalled();
  });

  it('should match snapshot when api response data is empty', () => {
    getCookie.mockReturnValue(false);
    const newProps = {
      ...defaultProps,
      termsConditions: {},
    };

    const { container } = setUp(newProps);

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when user is not logged in', () => {
    getCookie.mockReturnValue(false);
    const newProps = {
      ...defaultProps,
      termsConditions: {
        'en-HK': 'Terms and conditions',
        'zh-HK': 'Terms in Chinese',
      },
    };

    const { container } = setUp(newProps);

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when user is not logged in and changes the language tab', () => {
    getCookie.mockReturnValue(false);
    const newProps = {
      ...defaultProps,
      termsConditions: {
        'en-HK': 'Terms and conditions',
        'zh-HK': 'Terms in Chinese',
      },
    };

    const { container, getByTestId } = setUp(newProps);

    fireEvent.click(getByTestId('en-HK-tab'));

    expect(container).toMatchSnapshot();
  });

  it('should go to previous page on go back button click', () => {
    getCookie.mockReturnValue(false);
    const { getByTestId } = setUp();
    const backButton = getByTestId('btn-back-to-previous-page');

    fireEvent.click(backButton);

    expect(Router.back).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot when user has already accepted terms', () => {
    getCookie.mockReturnValue(true);
    const newProps = {
      ...defaultProps,
      termsConditions: {
        'en-HK': 'Terms and conditions',
        'zh-HK': 'Terms in Chinese',
      },
      alreadyAcceptedTerms: true,
    };

    const { container } = setUp(newProps);

    expect(container).toMatchSnapshot();
  });

  it('should dispatch acceptTerms & setLocal when terms is accepted', () => {
    jest.spyOn(Object.getPrototypeOf(window.localStorage), 'getItem');

    getCookie.mockReturnValue(true);
    const newProps = {
      ...defaultProps,
      termsConditions: {
        'en-HK': 'Terms and conditions',
        'zh-HK': 'Terms in Chinese',
      },
      alreadyAcceptedTerms: false,
    };

    const { container, getByTestId } = setUp(newProps);

    const agreeTermsCheckbox = getByTestId('checkbox-terms').querySelector(
      'input',
    );

    const agreeEdmCheckbox = getByTestId('checkbox-edm').querySelector('input');

    fireEvent.click(agreeTermsCheckbox);

    const acceptButton = getByTestId('btn-accept-terms');

    fireEvent.click(acceptButton);

    expect(container).toMatchSnapshot();
    expect(newProps.acceptTerms).toHaveBeenCalledWith(false);

    fireEvent.click(agreeEdmCheckbox);
    fireEvent.click(acceptButton);
    expect(newProps.acceptTerms).toHaveBeenCalledWith(true);
  });

  it('should logout when user disagrees with terms and logs out', () => {
    getCookie.mockReturnValue(true);
    const newProps = {
      ...defaultProps,
      termsConditions: {
        'en-HK': 'Terms and conditions',
        'zh-HK': 'Terms in Chinese',
      },
      alreadyAcceptedTerms: false,
    };

    const { getByTestId } = setUp(newProps);
    const logOutBtn = getByTestId('btn-logout');
    window.open = jest.fn();

    const localStorageMock = (function() {
      return {
        clear: jest.fn(),
      };
    })();
    Object.defineProperty(global, 'localStorage', { value: localStorageMock });

    fireEvent.click(logOutBtn);

    expect(window.open).toBeCalled();
    expect(window.open).toHaveBeenCalledWith('/logout', '_self');
    expect(window.localStorage.clear).toHaveBeenCalled();
  });
});
