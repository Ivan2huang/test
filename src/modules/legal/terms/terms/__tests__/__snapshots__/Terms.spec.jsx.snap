// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Terms component should dispatch acceptTerms & setLocal when terms is accepted 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-587"
  >
    <div>
      Typography component
      <span>
        type = style-2
      </span>
      <span>
        Terms & conditions
      </span>
    </div>
    <div
      class="MuiTabs-root makeStyles-tabContainer-580 makeStyles-tabContainer-584"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed"
        style="overflow: hidden;"
      >
        <div
          class="MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root makeStyles-firstTab-579 makeStyles-firstTab-583 MuiTab-textColorInherit Mui-selected"
            data-testid="en-HK-tab"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              <div>
                Typography component
                <span>
                  type = style-4
                </span>
                <span>
                  English
                </span>
              </div>
            </span>
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
            data-testid="zh-HK-tab"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              <div>
                Typography component
                <span>
                  type = style-4
                </span>
                <span>
                  中文 (繁體)
                </span>
              </div>
            </span>
          </button>
        </div>
        <span
          class="PrivateTabIndicator-root-612 PrivateTabIndicator-colorSecondary-614 MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-616 makeStyles-description-578 makeStyles-description-582"
      classes="[object Object]"
    >
      <div>
        Typography component
        <span>
          type = style-6
        </span>
        <span>
          <span>
            Terms and conditions
          </span>
        </span>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-617"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-630 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-631 Mui-checked MuiIconButton-colorPrimary"
          data-testid="checkbox-terms"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-633"
              data-indeterminate="false"
              type="checkbox"
              value="true"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <div>
            Typography component
            <span>
              type = style-6
            </span>
            <span>
              I Accept the terms and conditions
            </span>
          </div>
        </span>
      </label>
    </div>
    <div
      class="MuiBox-root MuiBox-root-682"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-630 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          data-testid="checkbox-edm"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-633"
              data-indeterminate="false"
              type="checkbox"
              value="false"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <div>
            Typography component
            <span>
              type = style-6
            </span>
            <span>
              (Optional) I do not want to receive health, wellness or insurance {promotional} (by selecting this option, you will not receive our exclusive offers)
            </span>
          </div>
        </span>
      </label>
    </div>
    <div
      class="MuiBox-root MuiBox-root-685 makeStyles-root-683 makeStyles-root-684"
      classes="[object Object]"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        data-testid="btn-accept-terms"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Agree and continue
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        data-testid="btn-logout"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Disagree and logout
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Terms component should match snapshot when api response data is empty 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-252"
  >
    <div>
      Typography component
      <span>
        type = style-2
      </span>
      <span>
        Terms & conditions
      </span>
    </div>
    <div
      class="MuiBox-root MuiBox-root-253 makeStyles-description-243 makeStyles-description-247"
      classes="[object Object]"
    >
      <div>
        Typography component
        <span>
          type = style-6
        </span>
        <span>
          <span />
        </span>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      data-testid="btn-back-to-previous-page"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Go back
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Terms component should match snapshot when user has already accepted terms 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-510"
  >
    <div>
      Typography component
      <span>
        type = style-2
      </span>
      <span>
        Terms & conditions
      </span>
    </div>
    <div
      class="MuiTabs-root makeStyles-tabContainer-503 makeStyles-tabContainer-507"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed"
        style="overflow: hidden;"
      >
        <div
          class="MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root makeStyles-firstTab-502 makeStyles-firstTab-506 MuiTab-textColorInherit Mui-selected"
            data-testid="en-HK-tab"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              <div>
                Typography component
                <span>
                  type = style-4
                </span>
                <span>
                  English
                </span>
              </div>
            </span>
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
            data-testid="zh-HK-tab"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              <div>
                Typography component
                <span>
                  type = style-4
                </span>
                <span>
                  中文 (繁體)
                </span>
              </div>
            </span>
          </button>
        </div>
        <span
          class="PrivateTabIndicator-root-535 PrivateTabIndicator-colorSecondary-537 MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-539 makeStyles-description-501 makeStyles-description-505"
      classes="[object Object]"
    >
      <div>
        Typography component
        <span>
          type = style-6
        </span>
        <span>
          <span>
            Terms and conditions
          </span>
        </span>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      data-testid="btn-back-to-previous-page"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Go back
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Terms component should match snapshot when user is logged in and has not accepted terms 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-10"
  >
    <div>
      Typography component
      <span>
        type = style-2
      </span>
      <span>
        Terms & conditions
      </span>
    </div>
    <div
      class="MuiBox-root MuiBox-root-11 makeStyles-description-1 makeStyles-description-5"
      classes="[object Object]"
    >
      <div>
        Typography component
        <span>
          type = style-6
        </span>
        <span>
          <span>
            Terms and conditions
          </span>
        </span>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-12"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-25 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          data-testid="checkbox-terms"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-28"
              data-indeterminate="false"
              type="checkbox"
              value="false"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <div>
            Typography component
            <span>
              type = style-6
            </span>
            <span>
              I Accept the terms and conditions
            </span>
          </div>
        </span>
      </label>
    </div>
    <div
      class="MuiBox-root MuiBox-root-80"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-25 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          data-testid="checkbox-edm"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-28"
              data-indeterminate="false"
              type="checkbox"
              value="false"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <div>
            Typography component
            <span>
              type = style-6
            </span>
            <span>
              (Optional) I do not want to receive health, wellness or insurance {promotional} (by selecting this option, you will not receive our exclusive offers)
            </span>
          </div>
        </span>
      </label>
    </div>
    <div
      class="MuiBox-root MuiBox-root-83 makeStyles-root-81 makeStyles-root-82"
      classes="[object Object]"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
        data-testid="btn-accept-terms"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Agree and continue
        </span>
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        data-testid="btn-logout"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Disagree and logout
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Terms component should match snapshot when user is not logged in 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-304"
  >
    <div>
      Typography component
      <span>
        type = style-2
      </span>
      <span>
        Terms & conditions
      </span>
    </div>
    <div
      class="MuiTabs-root makeStyles-tabContainer-297 makeStyles-tabContainer-301"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed"
        style="overflow: hidden;"
      >
        <div
          class="MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root makeStyles-firstTab-296 makeStyles-firstTab-300 MuiTab-textColorInherit"
            data-testid="en-HK-tab"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              <div>
                Typography component
                <span>
                  type = style-4
                </span>
                <span>
                  English
                </span>
              </div>
            </span>
          </button>
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
            data-testid="zh-HK-tab"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              <div>
                Typography component
                <span>
                  type = style-4
                </span>
                <span>
                  中文 (繁體)
                </span>
              </div>
            </span>
          </button>
        </div>
        <span
          class="PrivateTabIndicator-root-329 PrivateTabIndicator-colorSecondary-331 MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-333 makeStyles-description-295 makeStyles-description-299"
      classes="[object Object]"
    >
      <div>
        Typography component
        <span>
          type = style-6
        </span>
        <span>
          <span>
            Terms in Chinese
          </span>
        </span>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      data-testid="btn-back-to-previous-page"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Go back
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Terms component should match snapshot when user is not logged in and changes the language tab 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-381"
  >
    <div>
      Typography component
      <span>
        type = style-2
      </span>
      <span>
        Terms & conditions
      </span>
    </div>
    <div
      class="MuiTabs-root makeStyles-tabContainer-374 makeStyles-tabContainer-378"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed"
        style="overflow: hidden;"
      >
        <div
          class="MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root makeStyles-firstTab-373 makeStyles-firstTab-377 MuiTab-textColorInherit Mui-selected"
            data-testid="en-HK-tab"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              <div>
                Typography component
                <span>
                  type = style-4
                </span>
                <span>
                  English
                </span>
              </div>
            </span>
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
            data-testid="zh-HK-tab"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              <div>
                Typography component
                <span>
                  type = style-4
                </span>
                <span>
                  中文 (繁體)
                </span>
              </div>
            </span>
          </button>
        </div>
        <span
          class="PrivateTabIndicator-root-406 PrivateTabIndicator-colorSecondary-408 MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-410 makeStyles-description-372 makeStyles-description-376"
      classes="[object Object]"
    >
      <div>
        Typography component
        <span>
          type = style-6
        </span>
        <span>
          <span>
            Terms and conditions
          </span>
        </span>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      data-testid="btn-back-to-previous-page"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Go back
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
